### PR TITLE
fix cd path when registering a new package (fix #43)

### LINKED ISSUE #43: crash on PkgDev.register
My folder layout should be all standard.

```jl
julia> PkgDev.register("SuiteSparse")
INFO: Registering SuiteSparse at https://github.com/JuliaSparse/SuiteSparse.jl.git
INFO: Tagging SuiteSparse v0.0.1
ERROR: chdir METADATA: no such file or directory (ENOENT)
 [inlined code] from ./libuv.jl:68
 in cd(::String) at ./file.jl:39
 in cd(::PkgDev.Entry.##11#14{Base.LibGit2.GitRepo,String,VersionNumber,String,Bool,Array{Base.Pkg.Reqs.Line,1}}, ::String) at ./file.jl:47
 in write_tag_metadata(::Base.LibGit2.GitRepo, ::String, ::VersionNumber, ::String, ::Bool) at /home/kristoffer/.julia/v0.5/PkgDev/src/entry.jl:115
 in write_tag_metadata(::Base.LibGit2.GitRepo, ::String, ::VersionNumber, ::String) at /home/kristoffer/.julia/v0.5/PkgDev/src/entry.jl:110
 [inlined code] from ./dict.jl:780
 in (::PkgDev.Entry.##16#23{String,String,String})(::Base.LibGit2.GitRepo) at /home/kristoffer/.julia/v0.5/PkgDev/src/entry.jl:165
 in transact(::PkgDev.Entry.##16#23{String,String,String}, ::Base.LibGit2.GitRepo) at ./libgit2.jl:499
 in register(::String, ::String) at /home/kristoffer/.julia/v0.5/PkgDev/src/entry.jl:143
 in register(::String) at /home/kristoffer/.julia/v0.5/PkgDev/src/entry.jl:192
 in register(::String) at /home/kristoffer/.julia/v0.5/PkgDev/src/PkgDev.jl:32
 in eval(::Module, ::Any) at ./boot.jl:226
```

### PR DESCRIPTION
**Pull request by bicycle1885**
This will fix #43.

### CODE REVIEW

**Comment by tkelman** [comment]
might also fix https://github.com/JuliaLang/PkgDev.jl/issues/28, will have to test. xref #27.

**Comment by bicycle1885** [comment]
done

File: test/runtests.jl [hunk 1]
```diff
@@ -158,6 +158,12 @@ end"""
         f = PkgDev.freeable(io)
         @test any(f .== "Example") || contains(takebuf_string(io), "Example")
     end
+
+    @testset "testing package registration" begin
+        PkgDev.generate("GreatNewPackage", "MIT")
+        PkgDev.register("GreatNewPackage")
```
Comment (tkelman) [inline test/runtests.jl]: I think this will need git credentials of some kind

File: test/runtests.jl [hunk 2]
```diff
@@ -158,6 +158,12 @@ end"""
...
+        @test isdir(joinpath(pkgdir, "METADATA", "GreatNewPackage"))
```
Comment (tkelman) [inline test/runtests.jl]: maybe test for presence and non empty files

**Comment by bicycle1885** [comment]
```
INFO: Committing GreatNewPackage generated files
INFO: Registering GreatNewPackage at https://github.com/JuliaLang/GreatNewPackage.jl.git
INFO: Committing METADATA for GreatNewPackage
testing package registration: Error During Test
  Got an exception of type Base.LibGit2.Error.GitError outside of a @test
  GitError(Code:ENOTFOUND, Class:Config, Config value 'user.name' was not found)
   in macro expansion at ./libgit2/error.jl:98 [inlined]
   in default_signature(::Base.LibGit2.GitRepo) at ./libgit2/signature.jl:41
   in Base.LibGit2.Signature(::Base.LibGit2.GitRepo) at ./libgit2/signature.jl:24
   in commit(::Base.LibGit2.GitRepo, ::String) at ./libgit2/commit.jl:56
   in (::PkgDev.Entry.##20#28{String,String,String})(::Base.LibGit2.GitRepo) at /home/travis/.julia/v0.5/PkgDev/src/entry.jl:176
   in transact(::PkgDev.Entry.##20#28{String,String,String}, ::Base.LibGit2.GitRepo) at ./libgit2/libgit2.jl:501
   in register(::String, ::String) at /home/travis/.julia/v0.5/PkgDev/src/entry.jl:145
   in register(::String) at /home/travis/.julia/v0.5/PkgDev/src/entry.jl:194
   in register(::String) at /home/travis/.julia/v0.5/PkgDev/src/PkgDev.jl:36
   in macro expansion at /home/travis/.julia/v0.5/PkgDev/test/runtests.jl:170 [inlined]
   in macro expansion at ./test.jl:672 [inlined]
   in (::##1#2)(::String) at /home/travis/.julia/v0.5/PkgDev/test/runtests.jl:164
   in temp_pkg_dir(::##1#2, ::Bool) at /home/travis/.julia/v0.5/PkgDev/test/runtests.jl:15
   in temp_pkg_dir(::Function) at /home/travis/.julia/v0.5/PkgDev/test/runtests.jl:8
   in include_from_node1(::String) at ./loading.jl:426
   in process_options(::Base.JLOptions) at ./client.jl:262
   in _start() at ./client.jl:318
Test Summary:                | Error  Total
  testing package registration |     1      1
```

I may need to set the global user and email settings for Linux.

**Comment by tkelman** [comment]
so does the config dict not apply to both METADATA commits and the new repo? it seems like that could be changed?

File: test/runtests.jl [hunk 3]
```diff
@@ -20,6 +20,7 @@ end
 
 temp_pkg_dir() do pkgdir
 
+    #=
```
Comment (tkelman) [inline test/runtests.jl]: to be removed before merging, yeah?

Comment (bicycle1885) [inline test/runtests.jl]: Sure.

**Comment by bicycle1885** [comment]
> so does the config dict not apply to both METADATA commits and the new repo? it seems like that could be changed?

Sorry, I'm not sure what you mean. Each repository and METADATA read the global configuration set with the `git` command. Can you explain it more explicitly?

**Comment by tkelman** [comment]
In 922165d82e941ba237827909a807179d05825333 you tried setting a `config` dictionary. What was the reason that didn't work? Does the config dictionary get applied both to the generated package and any commits to metadata? If not, could it?

**Comment by bicycle1885** [comment]
If no global setting of git, we need to pass some information about the committer to this call: https://github.com/JuliaLang/PkgDev.jl/blob/e242d96556a9a9bd714b4a603f0715c5ae78eb21/src/entry.jl#L176, but it will need adding new parameters to `PkgDev.register`, which is I'm not sure a good approach or not.

File: test/runtests.jl [hunk 4]
```diff
@@ -158,6 +158,18 @@ end"""
         f = PkgDev.freeable(io)
         @test any(f .== "Example") || contains(takebuf_string(io), "Example")
     end
+
+    @testset "testing package registration" begin
+        if haskey(ENV, "CI") && lowercase(ENV["CI"]) == "true"
+            info("setting git global configuration")
+            run(`git config --global user.name "Julia Test"`)
+            run(`git config --global user.email test@julialang.org`)
+            run(`git config --global github.user JuliaTest`)
```
Comment (tkelman) [inline test/runtests.jl]: maybe a better approach would be to create a temporary directory, and do `withenv("HOME" => ...)` ?

Comment (bicycle1885) [inline test/runtests.jl]: Running `git config --local ...` inside a package repository may be better than modifying global configurations.

Comment (wildart) [inline test/runtests.jl]: `LibGit2.config`?

**Pull request merged**